lib/roger/renderer.rb

```diff
@@ -158,6 +158,10 @@ def render_file(path, options = {})
       end
 
       template = template(pn.to_s, nil)
+
+      # Once I understand I will update this comment
+      template_nesting.push(template)
```

Comment:
Apparently the context is not shared correctly, this was required but not necessarily a good fix!